File: hello.lua
--Define a "class" name HelloWorld
HelloWorld = {}
HelloWorld.__index = HelloWorld

function HelloWorld:new(name)
    local self = setmetatable({}, HelloWorld)
    self.name = name or "World"
    return self
end

function HelloWorld:greet()
    return "Hello, " .. self.name .. "!"
end

function HelloWorld:shout()
    return string.upper(self:greet())
end

local hw = HelloWorld:new("Lua")

print(hw:greet())
print(hw:shout())

File: hello.py
class HelloPython:
    def __init__(self, name: str = "World"):
        self.name = name

    def greet(self) -> str:
        return f"Hello, {self.name}!"

if __name__ == "__main__":
    hello = HelloPython()
    print(hello.greet())

